### PR TITLE
Add example for custom dataset

### PR DESCRIPTION
**Pull request by mihaimorariu**
This PR implements the example scripts for using a custom dataset.

### CODE REVIEW

**Comment by mihaimorariu** [comment]
@corochann: Can you please review?

**Comment by corochann** [comment]
Sorry for late review, let me review next week...

File: examples/custom_dataset/dataset_train.csv [hunk 1]
```diff
@@ -89,13 +89,3 @@ CC12CC1(C#N)C1CC12,-0.26750001311302185,0.02070000022649765
 CC(=O)C1OC1CC=O,-0.251800000667572,-0.04360000044107437
 Nc1cc(=O)cno1,-0.23770000040531158,-0.053700000047683716
 O=C1CC=CCC1O,-0.25519999861717224,-0.027300000190734863
```
Comment (corochann) [inline examples/custom_dataset/dataset_train.csv]: is this change necessary? if not necessary i want to keep it as is.

File: examples/custom_dataset/dataset_test.csv [hunk 2]
```diff
@@ -0,0 +1,11 @@
+SMILES,value1,value2
```
Comment (corochann) [inline examples/custom_dataset/dataset_test.csv]: We may create other new test dataset from QM9 instead of deleting train data, if necessary.

File: docs/source/tutorial.rst [hunk 3]
```diff
@@ -192,8 +192,8 @@ See output of ``python train_qm9.py --help``.
 
 Using your own dataset
 ========================
-You can use your own dataset in Chainer Chemistry.
-`example/own_dataset <https://github.com/pfnet-research/chainer-chemistry/tree/master/examples/own_dataset/>`_ shows an example.
+You can use your own dataset in Chainer Chemistry
+`example/custom_dataset <https://github.com/pfnet-research/chainer-chemistry/tree/master/examples/custom_dataset/>`_ shows an example.
```
Comment (corochann) [inline docs/source/tutorial.rst]: can you revert?

File: examples/test_examples.sh [hunk 4]
```diff
@@ -14,8 +14,8 @@ echo --- Testing QM9 ---
 cd qm9 && bash -x test_qm9.sh ${gpu} && cd ..
 
 # Own dataset
-echo --- Testing Own dataset ---
-cd own_dataset && bash -x test_own_dataset.sh ${gpu} && cd ..
+echo --- Testing on custom dataset ---
+cd custom_dataset && bash -x test_custom_dataset.sh ${gpu} && cd ..
```
Comment (corochann) [inline examples/test_examples.sh]: can you revert?

**Comment by mihaimorariu** [comment]
Thank you for your feedback. I have made the requested changes.

File: examples/own_dataset/plot.py [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+def save_evaluation_plot(x, y, metric, filename):
+    plt.figure()
+
+    sns.set()
+    ax = sns.barplot(y=x, x=y)
+
+    for n, (label, _y) in enumerate(zip(x, y)):
+        ax.annotate(
+            s='{:.3f}'.format(abs(_y)),
```
Comment (corochann) [inline examples/own_dataset/plot.py]: It did not work with matplotlib 3.0

**Comment by corochann** [comment]
Sorry for reviewing late!
